scripts/lib/json_parser.sh

```
@@ -0,0 +1,234 @@
+            if [[ -n "$current_extension" ]]; then
+                current_extension="${current_extension}${line}"
+            else
+                current_extension="$line"
+            fi
+        fi
+    done < "$json_file"
+}
+
+# Parse individual extension data string
+parse_extension_data() {
+    local extension_data="$1"
```

Shellcheck (0.10.0)

[warning] Variable was used as an array but is now assigned a string.

(SC2178)